195 correct the mapping on external resources links in supplementary information page

lingyun1010 opened the pull request:
This is a related PR for https://github.com/ebi-gene-expression-group/atlas-web-bulk/pull/204. The detailed rules are given by Silvie stated in the related issue. https://github.com/ebi-gene-expression-group/atlas-web-bulk/issues/195

lingyun1010 committed:
Decoupling experiment type from link to ega, ena &

geo external resources

lingyun1010 committed:
Decoupling experiment type from link to ega, ena &

geo external resources

lingyun1010 committed:
Decoupling experiment type from link to pride

lingyun1010 committed:
Decoupling experiment type from link to pride

lingyun1010 committed:
Remove unused imports

lingyun1010 committed:
Remove unused imports

lingyun1010 committed:
Remove abstract keyword

lingyun1010 committed:
Remove abstract keyword

lingyun1010 committed:
Decouple experiment type from linkToArrayExprss

lingyun1010 committed:
Decouple experiment type from linkToArrayExprss

lingyun1010 committed:
Correct IT test for all links classes

lingyun1010 committed:
Correct IT test for all links classes

ke4 commented on src/test/java/uk/ac/ebi/atlas/experimentpage/link/LinkToArrayExpressIT.java:
```diff
@@ -11,26 +11,11 @@
+    void emptyLinkIfExperimentNotOnArrayExpress() {
         var differentialExperiment = new ExperimentBuilder.DifferentialExperimentBuilder().build();
-        var subject = new LinkToArrayExpress.Differential();
+        var subject = new LinkToArrayExpress();
```
We could move it to a class variable as this is our subject in all of these test cases.

ke4 commented on src/test/java/uk/ac/ebi/atlas/experimentpage/link/LinkToArrayExpressIT.java:
```diff
@@ -68,10 +53,7 @@ void linkIfMicroarrayExperimentIsOnArrayExpress() {
 
     @RepeatedIfExceptionsTest(repeats = 5)
```
We probably don't need this anymore.

ke4 commented on src/test/java/uk/ac/ebi/atlas/experimentpage/link/LinkToEgaIT.java:
```diff
@@ -8,29 +8,14 @@
                         .withSecondaryAccessions(ImmutableList.of("EGA4545", "EGAS4546"))
                         .build();
-        var subject = new LinkToEga.ProteomicsBaseline();
+        var subject = new LinkToEga();
```
We can move this to a class variable and use that in the other test case, too.

ke4 commented on src/test/java/uk/ac/ebi/atlas/experimentpage/link/LinkToEgaIT.java:
```diff
@@ -8,29 +8,14 @@
-                .hasSize(1);
-    }
 
     @RepeatedIfExceptionsTest(repeats = 5)
```
We probably don't need this anymore.

ke4 commented on src/test/java/uk/ac/ebi/atlas/experimentpage/link/LinkToEgaIT.java:
```diff
@@ -42,48 +27,9 @@ void linksIfProteomicsBaselineExperimentIsOnEga() {
-                .hasSize(2);
-    }
-
     @RepeatedIfExceptionsTest(repeats = 5)
```
We probably don't need this anymore.

ke4 commented on src/test/java/uk/ac/ebi/atlas/experimentpage/link/LinkToEnaIT.java:
```diff
@@ -8,47 +8,14 @@
-                .hasSize(1);
-    }
 
     @RepeatedIfExceptionsTest(repeats = 5)
```
We probably don't need this anymore.

ke4 commented on src/test/java/uk/ac/ebi/atlas/experimentpage/link/LinkToGeoIT.java:
```diff
@@ -8,66 +8,14 @@
-                .hasSize(2);
-    }
 
     @RepeatedIfExceptionsTest(repeats = 5)
```
We probably don't need this anymore.

ke4 commented on src/test/java/uk/ac/ebi/atlas/experimentpage/link/LinkToEnaIT.java:
```diff
@@ -8,47 +8,14 @@
                         .withSecondaryAccessions(ImmutableList.of("ERP4545", "ERP4546"))
                         .build();
-        var subject = new LinkToEna.RnaSeqBaseline();
+        var subject = new LinkToEna();
```
We can move this to a class variable and use that in the other test case, too.

ke4 commented on src/test/java/uk/ac/ebi/atlas/experimentpage/link/LinkToGeoIT.java:
```diff
@@ -8,66 +8,14 @@
                         .withSecondaryAccessions(ImmutableList.of("GSE150361", "GSE5454"))
                         .build();
-        var subject = new LinkToGeo.Differential();
+        var subject = new LinkToGeo();
```
We can move this to a class variable and use that in the other test case, too.

ke4 reviewed (changes requested):
Good work. 👍 
Just a couple of simple comments.
Many thanks

ke4 committed:
[Chore] Core - Fix experiment design memory issue (#143)

* Remove experiment design field from Experiment

Temporary leave experimentDesign field on Experiment entity to pass the test. I am going to remove it in the last commit

* Refactor ExperimentDesignTable

* Remove experimentDesign field from Experiment entity

* Remove experimentDesign field from Experiment entity

lingyun1010 committed:
Refactor tests for all external resources

lingyun1010 committed:
Refactor tests for all external resources

ke4 reviewed (changes requested):
As I mentioned it on your related PR in `scxa` it looks like this branch is not based on the latest `atlas-web-core` changes.
Please fix that with either a merge or re-base.
Thanks

lingyun1010 committed:
Merge branch '195-correct-the-mapping-on-external-resources-links-in-supplementary-information-page' of https://github.com/ebi-gene-expression-group/atlas-web-core into 195-correct-the-mapping-on-external-resources-links-in-supplementary-information-page

ke4 commented:
@lingyun1010 Do you know why you have the same commits twice in your branch?
Could you eliminate it and clear this branch, please?

ke4 reviewed (approved):
LGTM!
